### PR TITLE
PROJ-176 Take into account new format of links

### PR DESCRIPTION
**Pull request by LucWollants**
### Fixed
- Take into account new format of links
---
Ticket: https://jira.uitdatabank.be/browse/PROJ-176

### CODE REVIEW

File: src/Insightly/Parser/LinkParser.php [hunk 1]
```diff
@@ -29,6 +29,18 @@ public static function parseToResult($data)
...
+        if (isset($data['LINK_OBJECT_NAME'], $data['LINK_OBJECT_ID'])) {
+            if ($data['LINK_OBJECT_NAME'] === 'Contact') {
+                $link->setContactId($data['LINK_OBJECT_ID']);
+            }
+
+            if ($data['LINK_OBJECT_NAME'] === 'Organisation') {
+                $link->setContactId($data['LINK_OBJECT_ID']);
+            }
+        }
+
```
Comment (bertramakers) [inline src/Insightly/Parser/LinkParser.php:43]: How about:

```
$linkObjectName = $data['LINK_OBJECT_NAME'] ?? '';
$linkObjectId = $data['LINK_OBJECT_ID'] ?? null;
if (($linkObjectName === 'Contact' || $linkObjectName === 'Organisation') && !is_null($linkObjectId)) {
    $link->setContactId($linkObjectId);
}
```

?

Comment (jdrieghe) [reply src/Insightly/Parser/LinkParser.php:43]: That doesn't really feel like a massive improvement over what @LucWollants suggested imho. Both are a workaround and both are "good enough" I think. The rest is very much preference.

Comment (bertramakers) [reply src/Insightly/Parser/LinkParser.php:43]: Should the second `$link->setContactId($data['LINK_OBJECT_ID']);` be some kind of `$link->setOrganisationId($data['LINK_OBJECT_ID']);` perhaps?

Comment (bertramakers) [reply src/Insightly/Parser/LinkParser.php:43]: If it's intentional `setContactId` twice I feel like a single `if` would better radiate that intent, that it's the same for both

Comment (LucWollants) [reply src/Insightly/Parser/LinkParser.php:43]: Good that you spotted this. This was a copy/paste error!

Comment (bertramakers) [reply src/Insightly/Parser/LinkParser.php:43]: Aha okay! Then 2 ifs make sense of course 😄

**Review by jdrieghe — APPROVED** [review]

**Pull request merged**